util/fipstools/acvp/acvptool/subprocess/ecdsa.go

```diff
@@ -167,8 +171,19 @@ func (e *ecdsa) Process(vectorSet []byte, m Transactable) (interface{}, error) {
 				if err != nil {
 					return nil, fmt.Errorf("signature generation failed for test case %d/%d: %s", group.ID, test.ID, err)
 				}
-				testResp.RHex = hex.EncodeToString(result[0])
-				testResp.SHex = hex.EncodeToString(result[1])
+				rHex := result[0]
+				sHex := result[1]
+				testResp.RHex = hex.EncodeToString(rHex)
+				testResp.SHex = hex.EncodeToString(sHex)
+				// Ask the subprocess to verify the generated signature for this test case.
+				ver_result, ver_err := m.Transact(e.algo+"/"+"sigVer", 1, []byte(group.Curve), []byte(group.HashAlgo), msg, qxHex, qyHex, rHex, sHex)
+				if ver_err != nil {
+					return nil, fmt.Errorf("After signature generation, signature verification failed for test case %d/%d: %s", group.ID, test.ID, ver_err)
+				}
+				// result[0] should be a single byte. The value should be one in this case.
```

Comment:
Why is that the case? May be you meant you meant `ver_result`?
Good call to make the subprocess check the output.

Reply:
Yes, self verification case.